metadata/src/utils/validation.rs

```diff
@@ -207,11 +224,34 @@ impl CachedHash {
 }
 
 /// Obtain the hash representation of a `scale_info::Type` identified by id.
+///
+/// Hashes of the outer enums (call, event, error) should be computed prior to this
+/// and passed in as the `outer_enum_hashes` argument. Whenever a type is encountered that
+/// is one of the outer enums, the procomputed hash is used instead of computing a new one.
+///
+/// The reason for this unintuitive behavior is that we sometimes want to trim the outer enum types
+/// beforehand to only include certain pallets, which affects their hash values.
 pub fn get_type_hash(
+    registry: &PortableRegistry,
+    id: u32,
+    outer_enum_hashes: &OuterEnumHashes,
+) -> Hash {
+    get_type_hash_recurse(registry, id, &mut HashMap::new(), outer_enum_hashes)
+}
+
+/// Obtain the hash representation of a `scale_info::Type` identified by id.
+fn get_type_hash_recurse(
     registry: &PortableRegistry,
     id: u32,
     cache: &mut HashMap<u32, CachedHash>,
-) -> [u8; HASH_LEN] {
+    outer_enum_hashes: &OuterEnumHashes,
+) -> Hash {
+    // If the type is part of `precomputed` hashes, return the precomputed hash instead:
```

Comment:
perhaps, tweak this comment now when "precomputed hashes" are called "outer_enum_hashes"?